query_optimizer/ExecutionGenerator.hpp

```diff
@@ -368,6 +384,13 @@ class ExecutionGenerator {
   QueryPlan *execution_plan_;  // A part of QueryHandle.
   serialization::QueryContext *query_context_proto_;  // A part of QueryHandle.
 
+#ifdef QUICKSTEP_DISTRIBUTED
+  serialization::CatalogDatabase *catalog_database_cache_proto_;  // A part of QueryHandle.
+
+  // Used to bookkeep relation ids for 'catalog_database_cache_proto_'.
+  std::unordered_set<relation_id> referenced_relation_ids_;
```

Comment:
I try to remove duplicate `relation_id`s, if any, i.e., a self-join query. So `vector` is not a choice. Also, we don't need a key-value map, so no `unordered_map`.